pkg/yamlmeta/map_key_overrides.go

```diff
@@ -0,0 +1,32 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package yamlmeta
+
+import (
+	"github.com/k14s/ytt/pkg/orderedmap"
+)
+
+type overrideMapKeys struct{}
+
+// Visit if `node` is a Map, among its MapItem's that have duplicate keys, removes all but the last.
+// This visitor always returns `nil`
+func (r *overrideMapKeys) Visit(node Node) error {
+	mapNode, isMap := node.(*Map)
+	if !isMap {
+		return nil
+	}
+
+	lastItems := orderedmap.NewMap()
```

Comment:
the only concern i have is that orderedmap throws away MapItem metadata. not sure how important this is here -- it's a little weird because you might be working with DocumentSet and suddently your MapItems get rewritten in a lossy way when calling OverrideMapKeys.

Reply:
But, in that case, the `value` in the `orderedmap.Map` _is_ a `yamlmeta.MapItem`... so, meta is preserved. Or am I missing something?

Reply:
oh jeez. thats very sneaky.